src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,108 @@
-import React from 'react';
+import cn from 'classnames';
+import React, { useContext, useState } from 'react';
+import { PostContext } from './context/PostContext';
+import { Error } from '../types/Error';
+import { Comment } from '../types/Comment';
+import { postComment } from '../api/comment';
+import { Post } from '../types/Post';
+
+type Props = {
+  selectedPost: Post | null,
+};
+
+export const NewCommentForm: React.FC<Props> = React.memo(({
+  selectedPost,
+}) => {
+  const {
+    setError,
+    setComments,
+    isLoading,
+    setIsLoading,
+  } = useContext(PostContext);
+
+  const [authorName, setAuthorName] = useState('');
+  const [isNameError, setIsNameError] = useState(false);
+
+  const [authorEmail, setAuthorEmail] = useState('');
+  const [isEmailError, setIsEmailError] = useState(false);
+
+  const [commentText, setCommentText] = useState('');
+  const [isCommentError, setIsCommentError] = useState(false);
```

Comment:
1) since author fields are one entity it's better to use an object (the same is for errors)
2) move initial states outside the component

```
const initialCommentState = {
  authorName: '',
  authorEmail: '',
  authorComment: '',
}

const initialErrorState = {
  isNameError: false,
  isEmailError: false,
  isCommentError: false,
}
```
```suggestion
  const [comment, setComment] = useState(initialCommentState);
  const [error, setError] = useState(initialErrorState)
```

Reply:
isn't fixed